### PR TITLE
Darren status recyclerview changes

### PR DESCRIPTION
**Pull request by DarrenPoh**
removed all status xmls, and added 1 general one

re-coded StatusAdapter.java

### CODE REVIEW

File: app/src/main/java/com/compscieddy/meetinthemiddle/HomeActivity.java [hunk 1]
```diff
@@ -60,12 +60,18 @@
   private GoogleApiClient mGoogleApiClient;
   private Marker mCurrentMarker;
 
-  @Bind(R.id.map) MapView mMapView;
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/HomeActivity.java:62]: Put all on same line - I know, it's annoying how Android Studio automatically adds a new line

File: app/src/main/java/com/compscieddy/meetinthemiddle/StatusAdapter.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    LayoutInflater layoutInflater = LayoutInflater.from(mContext);
+    View itemView = layoutInflater.inflate(R.layout.item_status, parent, false);
+    return new StatusHolder(itemView);
+  }
+
+  @Override
+  public void onBindViewHolder(StatusAdapter.StatusHolder holder, int position) {
+    //Placeholder text for now
+
+    if (position == CASINO) {
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/StatusAdapter.java]: I would use a switch/case here

File: app/src/main/java/com/compscieddy/meetinthemiddle/StatusAdapter.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      holder.statusImageView.setImageResource(R.drawable.ic_casino_white_24dp);
+    } else if (position == FITNESS) {
+      holder.statusImageView.setImageResource(R.drawable.ic_fitness_center_white_24dp);
+    } else if (position == MOVIES) {
+      holder.statusImageView.setImageResource(R.drawable.ic_local_movies_white_24dp);
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/StatusAdapter.java]: also, only set the imageResourceId = R.drawable.ic_local....

File: app/src/main/java/com/compscieddy/meetinthemiddle/StatusAdapter.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    } else {
+      holder.statusImageView.setImageResource(R.drawable.ic_local_hotel_white_24dp);
+    }
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/StatusAdapter.java:63]: Then, after the switch/case, you can do a single holder.statusImageView.setImageResource(imageResourceId)

File: app/src/main/java/com/compscieddy/meetinthemiddle/StatusAdapter.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+  }
+
+  @Override
+  public int getItemCount() {
+    return 4;
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/StatusAdapter.java]: Move this out into a constant, next to CASINO, MOVIES, GYM, etc

File: app/src/main/java/com/compscieddy/meetinthemiddle/StatusAdapter.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  public static final class StatusHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+    CircleImageView statusImageView;
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/StatusAdapter.java]: Once again, would use ImageView over CircleImageView

File: app/src/main/res/layout/activity_home.xml [hunk 7]
```diff
@@ -31,8 +31,22 @@
                 android:layout_width="match_parent"
                 android:layout_height="match_parent"
```
Comment (eddieberklee) [inline app/src/main/res/layout/activity_home.xml:35]: Can you delete the profile pictures actually? - I'll just redo them with the cropping

File: app/src/main/res/layout/activity_home.xml [hunk 8]
```diff
@@ -31,8 +31,22 @@
...
+
         </android.support.v7.widget.CardView>
 
+
+
+        <android.support.v7.widget.RecyclerView
+            android:id="@+id/status_recycler_view"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:layout_alignTop="@id/map_cardView"
```
Comment (eddieberklee) [inline app/src/main/res/layout/activity_home.xml]: I would do @+id/map_cardView here

File: app/src/main/res/layout/item_status.xml [hunk 9]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+             xmlns:app="http://schemas.android.com/apk/res-auto"
+             android:layout_width="48dp"
```
Comment (eddieberklee) [inline app/src/main/res/layout/item_status.xml]: Change this to wrap_content and just add a padding

File: app/src/main/res/layout/item_status.xml [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+             android:foreground="?android:attr/selectableItemBackground"
+             app:civ_border_color="#FF000000"
+             app:civ_border_width="2dp">
+
+    <de.hdodenhof.circleimageview.CircleImageView
+        android:id="@+id/status_image_view"
+        android:layout_width="40dp"
+        android:layout_height="40dp"
+        android:layout_gravity="center"
+        />
```
Comment (eddieberklee) [inline app/src/main/res/layout/item_status.xml]: Don't put this on a separate line please

File: app/src/main/res/layout/item_status.xml [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+
+
+</FrameLayout>
+
+    <!--
+        app:civ_border_color="#FF000000"
```
Comment (eddieberklee) [inline app/src/main/res/layout/item_status.xml]: delete

File: app/src/main/java/com/compscieddy/meetinthemiddle/StatusAdapter.java [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+  private static ClickListener mClickListener;
+
+  public static final int ITEM_COUNT = 4;
+
+  public static final int CASINO = 0;
+  public static final int FITNESS = 1;
+  public static final int MOVIES = 2;
+  public static final int SLEEPING = 3;
+
+  int imageResourceId;
```
Comment (eddieberklee) [inline app/src/main/java/com/compscieddy/meetinthemiddle/StatusAdapter.java]: Should be a local variable not a member variable

File: app/src/main/res/layout/activity_home.xml [hunk 13]
```diff
@@ -31,8 +31,22 @@
...
 
+
+
+        <android.support.v7.widget.RecyclerView
+            android:id="@+id/status_recycler_view"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:layout_alignTop="@+id/map_cardView"
+            android:layout_alignBottom="@+id/map_cardView"
+            android:layout_toRightOf="@id/map_cardView"
```
Comment (eddieberklee) [inline app/src/main/res/layout/activity_home.xml]: Missing a + here too

**Comment by DarrenPoh** [comment]
Created a placeholder status changer recycleview on home activity

**Pull request merged**